app-drawer/app-drawer.html

```diff
@@ -311,9 +337,22 @@
         this._setPosition(this.align);
       },
 
+      _focusHandler: function(event) {
+        if (!this.contains(Polymer.dom(event).rootTarget)) {
+          this._focusInsideDrawer();
+        }
+      },
+
+      _focusInsideDrawer: function() {
+        var child = Polymer.dom(this).querySelector(this.focusSelector);
+        if (child) {
+          child.focus();
+        }
+      },
+
       _scrimTapHandler: function() {
-        if (!this.persistent) {
-          this.opened = false;
+        if (this._isAboveContent()) {
```

Comment:
I think we can take out `_isAboveContent` check here since if the scrim is tappable which means is visible and so already above the content.